influence_benchmark/RL/EI.py

```diff
@@ -31,6 +31,4 @@ def _format_and_save_trajectories(self, selected_trajectories, trajectory_folder
 
             formatted_partial_trajs.append({"messages": messages_so_far, "num_hardcoded_msgs": num_hardcoded_msgs})
 
-        with open(trajectory_folder / "selected_trajectories.jsonl", "w", encoding="utf-8") as f:
-            for partial_traj in formatted_partial_trajs:
-                f.write(json.dumps(partial_traj) + "\n")
+        self._save_trajectories(formatted_partial_trajs, trajectory_folder)
```

Comment:
I think we should remove the saving from here, and just rename this method to `_format_trajectories` (both here and in KTO.py). And have the saving call in `base_iteration.py`